src/theory/shared_terms_database.cpp

```diff
@@ -55,8 +53,8 @@ void SharedTermsDatabase::setEqualityEngine(eq::EqualityEngine* ee)
     if (d_pfee == nullptr)
     {
       ProofNodeManager* pnm = d_env.getProofNodeManager();
-      d_pfeeAlloc.reset(
-          new eq::ProofEqEngine(d_satContext, d_userContext, *ee, pnm));
+      d_pfeeAlloc.reset(new eq::ProofEqEngine(
+          env.getContext(), env.getUserContext(), *ee, pnm));
```

Comment:
The following change should fix the PR:

```suggestion
          d_env.getContext(), d_env.getUserContext(), *ee, pnm));
```

Reply:
Thanks :)